docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md

```diff
@@ -0,0 +1,177 @@
+# Secret Management with GitOps
+
+GitOps primarily have git repositories in the center considered a source of truth for managing both infrastructure and application. This Infrastructure and application will require secured access to other resources of the system through secrets. Committing clear text secrets into git repositories is unacceptable even if the repositories are private to your team and organization. Teams need a secure way to handle secrets when using GitOps.
+
+There are many ways to manage secrets with GitOps and at high level can be categorized into:
+
+1. Encrypted Secrets in Git Repositories
+1. Reference to secrets stored in the external key vault
+
+## 1. Encrypted Secrets in Git Repositories
+
+In this approach, secrets are manually encrypted by developers using a public key and the custom Kubernetes controller running in the target cluster can only decrypt the key. Some popular tools for his approach are [Bitnami Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets), [Mozilla SOPS](https://github.com/mozilla/sops)
+
+- Secret changes are managed by making changes within the GitOps repository which provides great traceability of the changes made
+- All secrets can be rotated by making changes in GitOps, without accessing the cluster
+
+### [Bitnami Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets)
+
+Sealed secrets use asymmetric encryption to encrypt secrets. Kubernetes controller generates key pair(private-public) and stores the private key in etcd as Kubernetes secret. Developers use [Kubeseal CLI](https://github.com/bitnami-labs/sealed-secrets#public-key--certificate) to seal secrets before committing to the git repo.
+
+- [Support automatic key rotation](https://github.com/bitnami-labs/sealed-secrets#sealing-key-renewal) for the private key and can be used to enforce re-encryption of secrets
+  - Due to [automatic renewal of the sealing key](https://github.com/bitnami-labs/sealed-secrets#sealing-key-renewal), the key needs to be prefetched from the cluster or cluster set up to store the sealing key on renewal in a secondary location
+- Multi-tenancy support at the namespace level can be enforced by the controller
+- When sealing secrets developers needs a connection to the cluster control plane to fetch the public key or the public key has to be explicitly shared with the developer
+- Potential risk of secret checked into a git repo
+- If the private key in the cluster is lost for some reason all secrets need to be re-encrypted followed by a new key-pair generation
+- Does not scale with Multi-cluster here every cluster will require a controller having its own key pair
```

Comment:
Does "Multi" need to be capitalized?
Should there be a hyphen or a period between "cluster" and "here"?